app/api/routers/breeze_buddy/leads/rbac.py

```
@@ -94,19 +95,17 @@ def validate_lead_read_access(
-                status_code=status.HTTP_404_NOT_FOUND, detail=f"Lead not found"
-            )
+    # Check shop access. A null merchant_id marks a reseller-scoped lead: only
+    # reseller-role accounts (or admin, above) may read it — not every merchant
+    # sharing the reseller (PT-15).
+    if not merchant_scope_permitted(current_user, lead.merchant_id):
+        logger.warning(
+            f"User {current_user.username} attempted to {operation} lead {lead.id} "
+            f"for unauthorized merchant: {lead.merchant_id}"
+        )
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail=f"Lead not found"
```

Ruff (0.15.21)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)